### DISCUSSION TITLE
3W toolkit

### DISCUSSION THREAD

**Discussion by ricardoevvargas** [petrobras/3W#6] [Ideas]
**What can we do to make the 3W toolkit better?**

You can suggest anything related to its documentation, modules, interfaces, resources, etc.

Thanks for the contribution!

**Comment by FelipoDRS** [comment]
Good afternoon.

Thank you for your work, this dataset is very interesting. One improvement for the documentation I think is that it should be informed that the 10X class fault means the faulty transient of class X. It puzzled me the first time I analyzed the dataset and I don't think it is explicitly stated in the documentation.

I also think it should be informed explicitly that the pressure upstream of the PCK is 'P-MON-CKP' and the temperature downstream of the PCK is  'T-JUS-CKP'. We that speak Portuguese know that 'MON' is 'montante' and 'JUS' is 'jusante' but it may confuse someone that only speaks English, especially since the dataset also contains the tags 'P-JUS-CKGL' and 'T-JUS-CKGL' that look similar

These are my suggestions. Thank you a lot for your consideration.

**Reply by ricardoevvargas** [reply]
Dear @FelipoDRS, 

First of all, I thank you for your comment, which contributes to this discussion and to the 3W's development. I'm glad to know that you have been working with the 3W resources.

Yes, label 10X means class X transient. Currently, I and others working on this project understand that this is not the best strategy. Since this strategy is described in the paper that published the 3W ([here](https://doi.org/10.1016/j.petrol.2019.106223)) and other publicly available documents ([here ](https://github.com/ricardovvargas/3w_dataset/raw/master/docs/doctoral_thesis_ricardo_vargas.pdf) and [here](https://github.com/ricardovvargas/3w_dataset/raw/master/docs/master_degree_dissertation_bruno_carvalho.pdf)), I don't think it's worth tweaking the current 3W documentation. Also because we are working to generate and launch the 3W dataset's next version, which will have another labeling strategy. Until that happens, I hope this discussion helps others with this issue.

Yes, using 'MON' and 'JUS' in variable names is not good. This will also be improved in the 3W dataset's next version. Until that happens, we have this discussion and also succinct descriptions in the following 3W dataset's overview: [Baseline](overviews/_baseline/main.ipynb).

Thank you for your suggestions, which reinforce the need for improvements that we are working on. Feel encouraged to suggest further improvements, in this or other discussions.
